validator/src/main/java/com/amazon/aoc/helpers/CWMetricHelper.java

```diff
@@ -20,57 +20,130 @@
 import com.amazon.aoc.models.Context;
 import com.amazonaws.services.cloudwatch.model.Dimension;
 import com.amazonaws.services.cloudwatch.model.Metric;
+import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
 
 import java.nio.charset.StandardCharsets;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
+import java.util.*;
 
-/**
- * helper class for getting expected metrics from templates.
- */
+/** helper class for getting expected metrics from templates. */
 public class CWMetricHelper {
   private static final String DEFAULT_DIMENSION_NAME = "OTelLib";
   MustacheHelper mustacheHelper = new MustacheHelper();
 
   /**
    * get expected metrics from template with injecting context.
    *
-   * @param context        testing context
+   * @param context testing context
    * @param expectedMetric expected template
-   * @param caller         http caller, none caller, could be null
+   * @param caller http caller, none caller, could be null
    * @return list of metrics
    * @throws Exception when caller throws exception or template can not be found
    */
   public List<Metric> listExpectedMetrics(
-      Context context,
-      FileConfig expectedMetric,
-      ICaller caller
-  ) throws Exception {
+      Context context, FileConfig expectedMetric, ICaller caller) throws Exception {
     // call endpoint
     if (caller != null) {
       caller.callSampleApp();
     }
 
     // get expected metrics as yaml from config
-    String yamlExpectedMetrics = mustacheHelper.render(expectedMetric, context);
```

Comment:
Why does this change need to be made? It is repeated in both blocks of the `if/else` below.

Reply:
Isn't this line for removing this change, and adding it to the `if/else` blocks? + the line under initializes `yamlExpectedMetrics` so that it will compile correctly as it won't compile if it is declared only in the `if/else` block